llvm/include/llvm/ADT/APFloat.h

```diff
@@ -557,6 +559,14 @@ class IEEEFloat final : public APFloatBase {
 
   cmpResult compareAbsoluteValue(const IEEEFloat &) const;
 
+  /// returns the exponent in the format's radix
+  ExponentType getExponent() const;
+
+  /// returns the significand
+  APInt getSignificand() const;
+
+  /// decompose to an integer and exponent to radix 2
+  bool decomposeToIntegerAndPowerOf2(APInt &, int &) const;
```

Comment:
Maybe it is a limitation of APFloat, but this should return `std::optional<APInt, int>`.